src/irmin-pack/unix/chunked_suffix.ml

```diff
@@ -149,14 +192,32 @@ module Make (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
     let+ inventory = Inventory.open_ ~start_idx ~chunk_num ~open_chunk in
     { inventory }
 
+  let new_chunk ~root ~end_poff ~dead_header_size ~auto_flush_threshold
+      ~auto_flush_procedure t =
+    let open_chunk ~chunk_idx ~is_appendable =
+      match is_appendable with
+      | true ->
+          Ao_chunk.create_rw ~root ~chunk_idx ~overwrite:true
+            ~auto_flush_threshold ~auto_flush_procedure
+      | false ->
+          Ao_chunk.open_ro ~root ~chunk_idx ~end_poff ~dead_header_size
+            ~is_appendable
+    in
+    Inventory.add_new_appendable ~open_chunk t.inventory
+
   let open_ro ~root ~end_poff ~dead_header_size ~start_idx ~chunk_num =
     let open Result_syntax in
     let open_chunk = Ao_chunk.open_ro ~root ~end_poff ~dead_header_size in
     let+ inventory = Inventory.open_ ~start_idx ~chunk_num ~open_chunk in
     { inventory }
 
   let appendable_ao t = (Inventory.appendable t.inventory).ao
-  let end_poff t = appendable_ao t |> Ao.end_poff
+
+  let end_poff t =
+    let open Int63.Syntax in
+    let sum = ref Int63.zero in
+    Inventory.iter (fun c -> sum := !sum + Ao.end_poff c.ao) t.inventory;
+    !sum
```

Comment:
As said in https://github.com/mirage/irmin/pull/2115#discussion_r995922877 I think we should change how this works / how we use this value. I don't think a sum is correct. It "works" for a single chunk in our current code because the sum is the same as the value.

Reply:
With the current code it is a sum, and it works with several chunk files (the test on split passes). But I moved this into a new function `suffix_end_offset` and I will update accordingly to your PR.